CATs/KubernetesCluster.cat.rb

```diff
@@ -0,0 +1,475 @@
+
+This output displays the IP of the master server. Use your usual SSH connection method to initiate a SSH session on the master server.
+
+#### Authorized admin IPs
+
+Contains a list of IP addresses that have been authorized for full administrative access to the cluster.
+
+---
+
+### Actions
+
+#### Add Admin IP
+
+This action can be used to authorize an additional IP for full administrative access to the cluster.
+
+#### Update Autoscaling Range
+
+This action will modify the minimum and maximum number of cluster nodes. Although the action completes right away, please wait up to 30 minutes for the requested changes to happen in the background, especially when new servers are being launched.
+
+#### Install Hello app
+
+This action will install a basic Hello World web app onto the cluster.
+
+---"
+
+import "pft/mappings"
+import "pft/permissions"
+import "pft/account_utilities", as: "rs_acct"
+import "pft/server_templates_utilities", as: "rs_st"
+
+##################
+# Permissions    #
+##################
+permission "pft_general_permissions" do
+  like $permissions.pft_general_permissions
+end
+
+permission "pft_sensitive_views" do
+  like $permissions.pft_sensitive_views
+end
+
+##################
+# User inputs    #
+##################
+
+parameter "node_count_min" do
+  type "number"
+  label "Minimum Node Count"
+  category "Application"
+  description "Minimum number of cluster nodes."
+  min_value 1
+  max_value 30
+  default 3
+end
+
+parameter "node_count_max" do
+  type "number"
+  label "Maximum Node Count"
+  category "Application"
+  description "Maximum number of cluster nodes."
+  min_value 1
+  max_value 30
+  default 6
+end
+
+parameter "cloud" do
+  type "string"
+  label "Cloud"
+  category "Application"
+  description "Target cloud for this cluster."
+  allowed_values "AWS", "VMware"
+  default "AWS"
+end
+
+parameter "admin_ip" do
+  type "string"
+  label "Admin IP"
+  category "Application"
+  description "Allowed source IP for cluster administration. This IP address will have full access to the cluster."
+  allowed_pattern "^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$"
+  constraint_description "Please enter a single IP address. Additional IPs can be added after launch."
+end
+
+################################
+# Outputs returned to the user #
+################################
+
+output "master_ip" do
+  label "Master server IP"
+  category "Kubernetes"
+end
+
+output "dashboard_url" do
+  label "Launch Kubernetes dashboard"
+  category "Kubernetes"
+end
+
+output "admin_ips" do
+  label "Authorized admin IPs"
+  category "Kubernetes"
+end
+
+output "hello_url" do
+  label "View Hello app"
+  category "Kubernetes"
+end
+
+##############
+# MAPPINGS   #
+##############
+
+mapping "map_cloud" do
+  like $mappings.map_cloud
+end
+
+mapping "map_instancetype" do
+  like $mappings.map_instancetype
+end
+
+mapping "map_account" do {
+  "AWS" => {
+    "id" => "168969478693" },
+  "VMware" => {
+    "id" => null } }
+end
```

Comment:
Adam and I talked and we think the way to go is to use a credential for the cloud account ID and check for the credential in the bootstrap script. 
AWS will (generally) be handled automatically since the system by default creates the "AWS_ACCOUNT_NUMBER" credential.